fluent-bit-plugin-loki.yaml

```diff
@@ -0,0 +1,82 @@
+package:
+  name: fluent-bit-plugin-loki
+  version: 3.0.0
+  epoch: 0
+  description: The Fluent Bit loki plugin allows you to send your log or events to a Loki service.
+  copyright:
+    - license: AGPL-3.0-or-later
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - go
+  environment:
+    CGO_ENABLED: "1"
+
+pipeline:
+  - uses: git-checkout
+    with:
+      expected-commit: b4f7181c7aa9484e66976e8a933111a9b85ea8c2
+      repository: https://github.com/grafana/loki
+      tag: v${{package.version}}
+
+  - uses: go/bump
+    with:
+      deps: github.com/docker/docker@v25.0.5 golang.org/x/net@v0.23.0
+      replaces: go.etcd.io/bbolt=go.etcd.io/bbolt@v1.3.6
+  
+  - uses: go/build
```

Comment:
Trying to replicate this using go/build, but running into segmentation error
```
go build -ldflags "\
    -s -w \
    -X github.com/grafana/loki/v3/pkg/util/build.Branch=$(git rev-parse --abbrev-ref HEAD) \
    -X github.com/grafana/loki/v3/pkg/util/build.Version=3.0.0 \
    -X github.com/grafana/loki/v3/pkg/util/build.Revision=$(git rev-parse --short HEAD) \
    -X github.com/grafana/loki/v3/pkg/util/build.BuildUser=$(whoami)@$(hostname) \
    -X github.com/grafana/loki/v3/pkg/util/build.BuildDate=$(date -u +"%Y-%m-%dT%H:%M:%SZ")" \
    -tags netgo \
    -buildmode=c-shared \
    -o clients/cmd/fluent-bit/out_grafana_loki.so \
    ./clients/cmd/fluent-bit/
    ```

Reply:
Sent out a change to melange to plumb the `-buildmode` flag through `go/build`: https://github.com/chainguard-dev/melange/pull/1210

Once that's in (and we bump our deps here) this diff should do the trick:

```
diff --git a/fluent-bit-plugin-loki.yaml b/fluent-bit-plugin-loki.yaml
index f847f89b1..0f312d2a2 100644
--- a/fluent-bit-plugin-loki.yaml
+++ b/fluent-bit-plugin-loki.yaml
@@ -33,6 +33,7 @@ pipeline:
       packages: ./clients/cmd/fluent-bit/
       modroot: .
       output: out_grafana_loki.so
+      buildmode: c-shared
       ldflags: |
         -s -w
         -X github.com/grafana/loki/v3/pkg/util/build.Branch=$(git rev-parse --abbrev-ref HEAD)
@@ -40,7 +41,6 @@ pipeline:
         -X github.com/grafana/loki/v3/pkg/util/build.Revision=$(git rev-parse HEAD)
         -X github.com/grafana/loki/v3/pkg/util/build.BuildUser=$(whoami)@$HOSTNAME
         -X github.com/grafana/loki/v3/pkg/util/build.BuildDate=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
-        -buildmode=c-shared
       tags: netgo

   - runs: |
```